collect_app/src/test/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManagerTest.java

```diff
@@ -58,6 +60,7 @@
 
     private void stubSavedAccount(String accountName) {
         when(mockPreferences.get(PreferenceKeys.KEY_SELECTED_GOOGLE_ACCOUNT)).thenReturn(accountName);
+        stubAccount(accountName);
```

Comment:
This stub can be removed as it is redundant

Reply:
Okay, I get your point. The reason I had to add this here is that the tests were failing when the `credentials` didn't contain the account being checked for within the `getSelectedAccount` method fix I implemented.

Reply:
Which test case is failing?

Reply:
These tests were failing. 

![2018-08-18 23 03 19](https://user-images.githubusercontent.com/1509205/44305369-1e352f00-a33b-11e8-8acd-dc8ed0cc677f.png)

Yes, so mocking the `getAllAccounts` call allowed me to verify the new functionality adding in this PR and also verified that all the test cases relying on `getSelectedAccount` passed.

Reply:
Also when I merged in the new permissions code #2474 it broke most of the tests so I had to figure out a way to mock the way how the action on the listener was called.

Reply:
The new changes look good. But to ensure that credentials never return a blank account we should remove all `stubSavedAccount("");` from the test cases

Reply:
@jd-alexander there are still several `stubSavedAccount("");` in the tests. It would probably be best to change those to use a dummy account name, right?

Reply:
Actually, now I'm not so sure. It seems all those cases have to do with actually setting a blank account so it seems ok to me.

Reply:
I just did a check and I realized that it's not necessary to set the blank account since `getSelectedAccount` is returning an empty string so the tests are able to verify functionality without utilizing a blank account which would be the opposite of how the underlying `GooogleAccountCredential' API operates so I am removing them and pushing as we speak.

Reply:
Ahhh, thank you, that makes sense. 😊